### PR TITLE
DIALServer: Read DeviceId from the subsystem

### PR DESCRIPTION
**Pull request by HaseenaSainul**

### CODE REVIEW

File: DIALServer/DIALServer.h [hunk 1]
```diff
@@ -1273,6 +1273,30 @@ namespace Plugin {
             return (_deprecatedAPI);
         }
 
+        string DeviceId() const
+        {
+            string deviceId;
+            PluginHost::ISubSystem* subSystem = _service->SubSystems();
```
Comment (pwielders) [inline DIALServer/DIALServer.h:1279]: Be carefull, this is a leak. If you get the subSystem, it is an addref'ed interface make sure you release it!

Comment (HaseenaSainul) [reply DIALServer/DIALServer.h:1279]: done

Comment (pwielders) [reply DIALServer/DIALServer.h:1279]: Sorry, not completly, it is still leaking if the subsystem is retrieved (not a nullptr), it is addref'ed but if the SubSystem->IsActive() == false, it never gets released and thus it is leaking.. Suggest t remove the IsActive check from the equation and it is good to go...

**Review by pwielders — CHANGES_REQUESTED** [review]

**Review by pwielders — CHANGES_REQUESTED** [review]
It is still leaking if the system is not active, suggest to remove the IsActive call or split the if..

File: DIALServer/DIALServer.h (hunk 1 above)
Comment (HaseenaSainul) [reply DIALServer/DIALServer.h:1279]: done

**Comment by HaseenaSainul** [comment]
IsActive check is removed

**Review by pwielders — APPROVED** [review]

**Pull request merged**